apps/schematic/api/schematic_api/controllers/manifest_generation_controller_impl.py

```diff
@@ -0,0 +1,103 @@
+"""Manifest generation functions"""
+# pylint: disable=too-many-arguments
+
+from schematic import CONFIG  # type: ignore
+from schematic.manifest.generator import ManifestGenerator  # type: ignore
+
+from schematic_api.models.basic_error import BasicError
+from schematic_api.models.google_sheet_links import GoogleSheetLinks
+from schematic_api.controllers.utils import (
+    handle_exceptions,
+    get_access_token,
+    download_schema_file_as_jsonld,
+    InvalidValueError,
+)
+
+
+@handle_exceptions
+def generate_google_sheet_manifests(
+    schema_url: str,
+    asset_view_id: str,
+    add_annotations: bool,
+    dataset_id_array: list[str] | None,
+    manifest_title: str,
+    node_label_array: list[str] | None,
+    use_strict_validation: bool,
+    generate_all_manifests: bool,
+) -> tuple[GoogleSheetLinks | BasicError, int]:
+    """Generates a list of links to manifets in google sheet form
+
+    Args:
+        schema_url (str): The URL of the schema
+        dataset_id_array (list[str] | None): Use this to get the existing manifests in the
+          datasets. Must be of same type as the node_label_array, same order, and same length
+        asset_view_id (str): ID of the asset view
+        node_label_array (list[str] | None): The datatypes of the manifests to generate
+        add_annotations (bool): Whether or not annotatiosn get added to the manifest
+        manifest_title (str): Title of the manifest
+        use_strict_validation (bool): Whether or not to use google sheet strict validation
+        generate_all_manifests (bool): Will generate a manifest for all data types
+
+    Raises:
+        ValueError: When generate_all_manifests is true and either dataset_id_array or
+          node_label_array are provided
+        ValueError: When generate_all_manifests is false and node_label_array is not provided
+        ValueError: When generate_all_manifests is false and dataset_id_arrayy is provided,
+          but it doesn't match the length of node_label_array
+
+    Returns:
+        tuple[GoogleSheetLinks | BasicError, int]: A tuple
+           The first item is either the google sheet links of the manifests or an error object
+           The second item is the response status
+    """
+
+    if generate_all_manifests:
+        if dataset_id_array:
+            raise InvalidValueError(
+                "When generate_all_manifests is True dataset_id_array must be None",
+                {"dataset_id_array": dataset_id_array},
+            )
+        if node_label_array:
+            raise InvalidValueError(
+                "When generate_all_manifests is True node_label_array must be None",
+                {"node_label_array": node_label_array},
+            )
+        node_label_array = ["all_manifests"]
```

Comment:
here, if should be "all manifests" instead of "all_manifests". Otherwise schematic would error out.

Reply:
Yes you are right, I thought it was the other way around